src/abi_1_0.c

```diff
@@ -451,3 +518,55 @@ int fi_getinfo_1_2(uint32_t version, const char *node, const char *service,
 	return ret;
 }
 COMPAT_SYMVER(fi_getinfo_1_2, fi_getinfo, FABRIC_1.2);
+
+/*
+ * ABI 1.3
+ */
+__attribute__((visibility ("default"),EXTERNALLY_VISIBLE))
+void fi_freeinfo_1_3(struct fi_info_1_3 *info)
+{
+	fi_freeinfo((struct fi_info *) info);
+}
+COMPAT_SYMVER(fi_freeinfo_1_3, fi_freeinfo, FABRIC_1.3);
+
+__attribute__((visibility ("default"),EXTERNALLY_VISIBLE))
+struct fi_info_1_3 *fi_dupinfo_1_3(const struct fi_info_1_3 *info)
+{
+	struct fi_info *dup, *base;
+
+	if (!info)
+		return (struct fi_info_1_3 *) ofi_allocinfo_internal();
+
+	ofi_dup_attr(base, info);
+	if (base == NULL)
+		return NULL;
+
+	dup = fi_dupinfo(base);
```

Comment:
This only works for the case that new fields are added into `struct fi_info`. However, for new fields in `tx_attr`, `rx_attr`, `domain_attr` etc, the resulting `dup` would have those new fields filled with unknown values.

The CI failure with SHMEM is caused by this. It failed at checking `domain_attr` for rxm due to a randomly large number was seen by the provider as `max_ep_auth_key`. The SHMEM test was pre-built against an older version of libfabric.

Reply:
Thanks for finding this. Will fix!

Reply:
fi_dupinfo_1_1() and fi_dupinfo_1_2() need to be updated similarly.

Reply:
Hmm I think the are more problems that just this with the ABI compat. For example, the FABRIC_1.2 ABI compat uses a 1.1 domain attr definition which is wrong.

Reply:
Actually, may be OK. I may have to redo my compat support.

Reply:
Ok. I uploaded a patch to fix this.